Compile error on ppc64 - overloaded call is ambiguous

Trying to build protobuf 3.2.0rc2 for Fedora on ppc64 I get:
```
libtool: compile:  g++ -DHAVE_CONFIG_H -I. -I.. -DHAVE_PTHREAD=1 -DHAVE_ZLIB=1 -Wall -Wno-sign-compare -O2 -g -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector-strong --param=ssp-buffer-size=4 -grecord-gcc-switches -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -m64 -c google/protobuf/wire_format_lite.cc  -fPIC -DPIC -o google/protobuf/.libs/wire_format_lite.o
google/protobuf/wire_format_lite.cc: In instantiation of 'void google::protobuf::internal::WriteArray(const CType*, int, google::protobuf::io::CodedOutputStream*) [with CType = int]':
google/protobuf/wire_format_lite.cc:414:33:   required from here
google/protobuf/wire_format_lite.cc:384:27: error: call of overloaded 'EncodeFixedSizeValue(const int&, google::protobuf::uint8*&)' is ambiguous
       EncodeFixedSizeValue(a[i+j], ptr);
       ~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~
google/protobuf/wire_format_lite.cc:344:6: note: candidate: void google::protobuf::internal::{anonymous}::EncodeFixedSizeValue(float, google::protobuf::uint8*)
 void EncodeFixedSizeValue(float v, uint8* dest) {
      ^~~~~~~~~~~~~~~~~~~~
google/protobuf/wire_format_lite.cc:348:6: note: candidate: void google::protobuf::internal::{anonymous}::EncodeFixedSizeValue(double, google::protobuf::uint8*)
 void EncodeFixedSizeValue(double v, uint8* dest) {
      ^~~~~~~~~~~~~~~~~~~~
google/protobuf/wire_format_lite.cc:352:6: note: candidate: void google::protobuf::internal::{anonymous}::EncodeFixedSizeValue(google::protobuf::uint32, google::protobuf::uint8*)
 void EncodeFixedSizeValue(uint32 v, uint8* dest) {
      ^~~~~~~~~~~~~~~~~~~~
google/protobuf/wire_format_lite.cc:356:6: note: candidate: void google::protobuf::internal::{anonymous}::EncodeFixedSizeValue(google::protobuf::uint64, google::protobuf::uint8*)
 void EncodeFixedSizeValue(uint64 v, uint8* dest) {
      ^~~~~~~~~~~~~~~~~~~~
google/protobuf/wire_format_lite.cc:368:6: note: candidate: void google::protobuf::internal::{anonymous}::EncodeFixedSizeValue(bool, google::protobuf::uint8*)
 void EncodeFixedSizeValue(bool v, uint8* dest) {
      ^~~~~~~~~~~~~~~~~~~~
google/protobuf/wire_format_lite.cc: In instantiation of 'void google::protobuf::internal::WriteArray(const CType*, int, google::protobuf::io::CodedOutputStream*) [with CType = long int]':
google/protobuf/wire_format_lite.cc:419:33:   required from here
google/protobuf/wire_format_lite.cc:384:27: error: call of overloaded 'EncodeFixedSizeValue(const long int&, google::protobuf::uint8*&)' is ambiguous
       EncodeFixedSizeValue(a[i+j], ptr);
       ~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~
google/protobuf/wire_format_lite.cc:344:6: note: candidate: void google::protobuf::internal::{anonymous}::EncodeFixedSizeValue(float, google::protobuf::uint8*)
 void EncodeFixedSizeValue(float v, uint8* dest) {
      ^~~~~~~~~~~~~~~~~~~~
google/protobuf/wire_format_lite.cc:348:6: note: candidate: void google::protobuf::internal::{anonymous}::EncodeFixedSizeValue(double, google::protobuf::uint8*)
 void EncodeFixedSizeValue(double v, uint8* dest) {
      ^~~~~~~~~~~~~~~~~~~~
google/protobuf/wire_format_lite.cc:352:6: note: candidate: void google::protobuf::internal::{anonymous}::EncodeFixedSizeValue(google::protobuf::uint32, google::protobuf::uint8*)
 void EncodeFixedSizeValue(uint32 v, uint8* dest) {
      ^~~~~~~~~~~~~~~~~~~~
google/protobuf/wire_format_lite.cc:356:6: note: candidate: void google::protobuf::internal::{anonymous}::EncodeFixedSizeValue(google::protobuf::uint64, google::protobuf::uint8*)
 void EncodeFixedSizeValue(uint64 v, uint8* dest) {
      ^~~~~~~~~~~~~~~~~~~~
google/protobuf/wire_format_lite.cc:368:6: note: candidate: void google::protobuf::internal::{anonymous}::EncodeFixedSizeValue(bool, google::protobuf::uint8*)
 void EncodeFixedSizeValue(bool v, uint8* dest) {
      ^~~~~~~~~~~~~~~~~~~~
google/protobuf/wire_format_lite.cc:364:6: warning: 'void google::protobuf::internal::{anonymous}::EncodeSFixedSizeValue(google::protobuf::int64, google::protobuf::uint8*)' defined but not used [-Wunused-function]
 void EncodeSFixedSizeValue(int64 v, uint8* dest) {
      ^~~~~~~~~~~~~~~~~~~~~
google/protobuf/wire_format_lite.cc:360:6: warning: 'void google::protobuf::internal::{anonymous}::EncodeSFixedSizeValue(google::protobuf::int32, google::protobuf::uint8*)' defined but not used [-Wunused-function]
 void EncodeSFixedSizeValue(int32 v, uint8* dest) {
      ^~~~~~~~~~~~~~~~~~~~~
make[2]: *** [Makefile:3974: google/protobuf/wire_format_lite.lo] Error 1
```
Full log here: https://kojipkgs.fedoraproject.org//work/tasks/1573/17411573/build.log
x86_64, i386, and ppc64le build okay - https://koji.fedoraproject.org/koji/taskinfo?taskID=17411556

I'll note that on x86_64 I get these possibly related warnings:
```
google/protobuf/wire_format_lite.cc:368:6: warning: 'void google::protobuf::internal::{anonymous}::EncodeFixedSizeValue(bool, google::protobuf::uint8*)' defined but not used [-Wunused-function]
 void EncodeFixedSizeValue(bool v, uint8* dest) {
      ^~~~~~~~~~~~~~~~~~~~
google/protobuf/wire_format_lite.cc:364:6: warning: 'void google::protobuf::internal::{anonymous}::EncodeSFixedSizeValue(google::protobuf::int64, google::protobuf::uint8*)' defined but not used [-Wunused-function]
 void EncodeSFixedSizeValue(int64 v, uint8* dest) {
      ^~~~~~~~~~~~~~~~~~~~~
google/protobuf/wire_format_lite.cc:360:6: warning: 'void google::protobuf::internal::{anonymous}::EncodeSFixedSizeValue(google::protobuf::int32, google::protobuf::uint8*)' defined but not used [-Wunused-function]
 void EncodeSFixedSizeValue(int32 v, uint8* dest) {
      ^~~~~~~~~~~~~~~~~~~~~
google/protobuf/wire_format_lite.cc:356:6: warning: 'void google::protobuf::internal::{anonymous}::EncodeFixedSizeValue(google::protobuf::uint64, google::protobuf::uint8*)' defined but not used [-Wunused-function]
 void EncodeFixedSizeValue(uint64 v, uint8* dest) {
      ^~~~~~~~~~~~~~~~~~~~
google/protobuf/wire_format_lite.cc:352:6: warning: 'void google::protobuf::internal::{anonymous}::EncodeFixedSizeValue(google::protobuf::uint32, google::protobuf::uint8*)' defined but not used [-Wunused-function]
 void EncodeFixedSizeValue(uint32 v, uint8* dest) {
      ^~~~~~~~~~~~~~~~~~~~
google/protobuf/wire_format_lite.cc:348:6: warning: 'void google::protobuf::internal::{anonymous}::EncodeFixedSizeValue(double, google::protobuf::uint8*)' defined but not used [-Wunused-function]
 void EncodeFixedSizeValue(double v, uint8* dest) {
      ^~~~~~~~~~~~~~~~~~~~
google/protobuf/wire_format_lite.cc:344:6: warning: 'void google::protobuf::internal::{anonymous}::EncodeFixedSizeValue(float, google::protobuf::uint8*)' defined but not used [-Wunused-function]
 void EncodeFixedSizeValue(float v, uint8* dest) {
      ^~~~~~~~~~~~~~~~~~~~
```

--- a/src/google/protobuf/wire_format_lite.cc
+++ b/src/google/protobuf/wire_format_lite.cc
@@ -357,11 +357,11 @@ void EncodeFixedSizeValue(uint64 v, uint8* dest) {
   WireFormatLite::WriteFixed64NoTagToArray(v, dest);
 }
 
-void EncodeSFixedSizeValue(int32 v, uint8* dest) {
+void EncodeFixedSizeValue(int32 v, uint8* dest) {
   WireFormatLite::WriteSFixed32NoTagToArray(v, dest);
 }
 
-void EncodeSFixedSizeValue(int64 v, uint8* dest) {
+void EncodeFixedSizeValue(int64 v, uint8* dest) {
   WireFormatLite::WriteSFixed64NoTagToArray(v, dest);
 }
 

